sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -75,9 +76,9 @@ class JoinSuite extends QueryTest with BeforeAndAfterEach {
       ("SELECT * FROM testData FULL OUTER JOIN testData2 WHERE key = 2", classOf[CartesianProduct]),
       ("SELECT * FROM testData JOIN testData2 WHERE key > a", classOf[CartesianProduct]),
       ("SELECT * FROM testData FULL OUTER JOIN testData2 WHERE key > a", classOf[CartesianProduct]),
-      ("SELECT * FROM testData JOIN testData2 ON key = a", classOf[ShuffledHashJoin]),
```

Comment:
Can we keep the original test case `ShuffledHashJoin`? We an do that by turn off/on the `AUTO_SORTMERGEJOIN`.